operators/inpaint_area_brush.py

```diff
@@ -1,62 +1,32 @@
-    bl_idname = 'dream_textures.inpaint_area_stroke'
-    bl_label = "Mark Area for Inpainting"
-    bl_description = "Marks the area for inpainting by setting the alpha to 0 while keeping RGB values"
-    bl_options = {'REGISTER'}
-
-    firing_mode: bpy.props.IntProperty()
-    alpha_mode: bpy.props.BoolProperty(default=False)
-
-    def invoke(self, context, event):
-        global is_painting
-        global last_stroke
-        global time
-        if self.firing_mode == 0: # PRESS
-            is_painting = True
-        elif self.firing_mode == 2: # RELEASE
-            is_painting = False
-            last_stroke = None
-            time = 0
-        elif self.firing_mode == 1 and is_painting: # MOVE
-            original_tool = context.workspace.tools.from_space_image_mode('PAINT').idname
-        
-            bpy.ops.paint.brush_select(image_tool='DRAW', toggle=False)
-            brush = bpy.data.brushes["TexDraw"]
-            brush.use_pressure_strength = False
-            brush.blend = 'ADD_ALPHA' if self.alpha_mode else 'ERASE_ALPHA'
-            context.tool_settings.image_paint.brush = brush
-            stroke = {
-                "name": "stroke",
-                "mouse": (event.mouse_x, event.mouse_y - context.scene.tool_settings.unified_paint_settings.size),
-                "mouse_event": (0,0),
-                "pen_flip" : True,
-                "is_start": True if last_stroke is None else False,
-                "location": (0, 0, 0),
-                "size": context.scene.tool_settings.unified_paint_settings.size,
-                "pressure": 1,
-                "x_tilt": 0,
-                "y_tilt": 0,
-                "time": float(time)
-            }
-            if last_stroke is not None:
-                with context.temp_override():
-                    bpy.ops.paint.image_paint(stroke=[last_stroke, stroke], mode='NORMAL')
-            last_stroke = stroke
-            time += 1
-            bpy.ops.wm.tool_set_by_id(name=original_tool)
-        
-        return {"FINISHED"}
-
-    def execute(self, context):
-        return {"FINISHED"}
+reset_blend_mode = 'MIX'
+reset_curve_preset = 'CUSTOM'
+reset_strength = 1.0
+class InpaintAreaBrushActivated(bpy.types.GizmoGroup):
+    bl_idname = "dream_textures.inpaint_area_brush_activated"
+    bl_label = "Inpaint Area Brush Activated"
+    bl_space_type = 'IMAGE_EDITOR'
+    bl_context_mode = 'PAINT'
+    bl_region_type = 'WINDOW'
+
+    def setup(self, context):
+        global reset_blend_mode
+        global reset_curve_preset
+        global reset_strength
+        reset_blend_mode = bpy.data.brushes["TexDraw"].blend
+        reset_curve_preset = bpy.data.brushes["TexDraw"].curve_preset
+        reset_strength = bpy.data.brushes["TexDraw"].strength
+        def set_blend():
+            bpy.data.brushes["TexDraw"].blend = "ERASE_ALPHA"
+            bpy.data.brushes["TexDraw"].curve_preset = "CONSTANT"
+            bpy.data.brushes["TexDraw"].strength = 1.0
```

Comment:
Inpaint can use the wrong tool if the draw tool wasn't selected
```suggestion
            bpy.data.brushes["TexDraw"].strength = 1.0
            bpy.ops.paint.brush_select(image_tool='DRAW', toggle=False)
```